src/os/bluestore/BlueStore.cc

```diff
@@ -3132,7 +3132,9 @@ int BlueStore::_decompress(bufferlist& source, bufferlist* result)
     derr << __func__ << " can't load decompressor " << chdr.type << dendl;
     r = -EIO;
   } else {
-    r = compressor->decompress(i, *result);
+    bufferlist t;
```

Comment:
Shouldn't we provide decompress with two iterators ( begin,end) instead to avoid data copying?